tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -1557,3 +1621,156 @@ def _test_memory_based_scheduling_enabled_true(
     # check RealMemory overridden via config file parameter
     assert_that(slurm_commands.get_node_attribute("queue1-dy-ondemand1-i3-1", "Memory")).is_equal_to("31400")
     # TODO: Add functional tests for memory-based scheduling
+
+
+def _test_scontrol_reboot_idle_nodes(
+    remote_command_executor,
+    slurm_commands,
+):
+    """Test scontrol reboot with idle nodes."""
+
+    # Get nodes and check that they are idle
+    nodes_in_queue = slurm_commands.get_compute_nodes("queue1")
+    assert_compute_node_states(slurm_commands, nodes_in_queue, "idle")
+
+    # Reboot nodes according to this logic:
+    # - 1 static idle node
+    # - 1 static idle node asap
+    # - 1 dynamic idle node
+    # - 1 dynamic idle node asap
+    static_nodes, dynamic_nodes = get_partition_nodes(nodes_in_queue)
+    assert_that(len(static_nodes)).is_equal_to(2)
+    assert_that(len(dynamic_nodes)).is_equal_to(2)
+    slurm_commands.reboot_compute_node(static_nodes[0], False)
+    slurm_commands.reboot_compute_node(static_nodes[1], True)
+    slurm_commands.reboot_compute_node(dynamic_nodes[0], False)
+    slurm_commands.reboot_compute_node(dynamic_nodes[1], True)
+
+    # Check that nodes enter a reboot state
+    time.sleep(2)
+    assert_compute_node_states(slurm_commands, nodes_in_queue, "reboot^")
+
+    # Wait that nodes come back after a while, without having triggered clustermgtd
```

Comment:
It's not clear to me the relation between this comment and the actual check in line 1658. 
Why do we consider the absence of the log line `Found the following unhealthy static nodes` as a signal that clustermgtd has not been triggered?

Reply:
Because if it had been triggered (i.e. if the node had been marked as unhealthy), you would see it in the logs with `Found the following unhealthy static nodes (x1) : ['nodename'(..."`, which comes from [here](https://github.com/aws/aws-parallelcluster-node/blob/9b9e1145fe212fb7b079821453e969b3adb5ae23/src/slurm_plugin/clustermgtd.py#L734).